fhirstore/search/urlparser.py

```diff
@@ -0,0 +1,138 @@
+from collections import defaultdict
+import re
+
+# Format the url args to a dict
+def url_to_dict(url_args):
+    search_args = {key: url_args.getlist(key) for key in url_args.keys()}
+    return search_args
+
+
+# Parse commas in a (key,value)
+def parse_comma(key, value):
+    has_comma = "," in value
+    if has_comma:
+        return "multiple", {key: value.split(",")}
+    else:
+        return key, [value]
+
+
+# Process all the dict for possible commas
+def pre_process_params(url_args):
+    search_args = url_to_dict(url_args)
+    processed_params = defaultdict(list)
+    if search_args == {}:
+        processed_params = {}
+
+    for key, value in search_args.items():
+        if len(value) == 1:
+            parsed_key, parsed_value = parse_comma(key, value[0])
+            processed_params.update({parsed_key: parsed_value})
+        else:
+            for element in value:
+                parsed_key, parsed_value = parse_comma(key, element)
+                if parsed_key == "multiple":
+                    processed_params[parsed_key] = parsed_value
+                # /                    processed_params["multiple"][sub_key].append(parsed_dict["multiple"][sub_key])
+                else:
+                    processed_params[parsed_key].append(parsed_value[0])
+    return processed_params
+
+
+class SearchArguments:
+    def __init__(self, url_args, resource_type):
+        self.args = pre_process_params(url_args)
+        self.resource_type = resource_type
+
+        # core args
+        self.core_args = {}
+        self.sort = None
+
+        # formatting args
+        self.elements = None
+        self.include = False
+
+        # meta args
+        self.summary = False
+        self.is_summary_count = False
+        self.offset = 0
+        self.result_size = 100
+
+
+class UrlParser:
+    def __init__(self):
+        self.core_args = {}
+
+    def sort_params(self, args):
+        has_sort = None
+        if "_sort" in args:
+            has_sort = args["_sort"]
+        elif "multiple" in args:
+            has_sort = args["multiple"].get("_sort", None)
+        # if there is a sorting argument, process it to handle a change of sorting order
+        if has_sort:
+            sorting_params = []
+            for argument in has_sort:
+                # find a "-" before the argument. It indicates a sorting order different from default
+                has_minus = re.search(r"^-(.*)", argument)
```

Comment:
I don't think you need a regex here, maybe argument.startswith("-") is sufficient

Reply:
And you can remove this heading "-" next with argument[1:], not so clean but I still prefer that to re ^^